samples/subsys/logging/kernel_event_logger/src/main.c

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2017 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel_structs.h>
+#include <logging/kernel_event_logger.h>
+#include <kernel_structs.h>
+#include <misc/printk.h>
+#include <misc/util.h>
+#include <zephyr.h>
+
+K_THREAD_STACK_DEFINE(threadA_stack, 1024);
+K_THREAD_STACK_DEFINE(threadB_stack, 1024);
+K_THREAD_STACK_DEFINE(event_logger_stack, 4096);
+
+K_SEM_DEFINE(sem_sync, 0, 1);	/* starts off "not available" */
+
+static struct k_thread threadA_data;
+static struct k_thread threadB_data;
+static struct k_thread event_logger_thread_data;
+static u32_t timestamp;
+char thread_event_type[][12] = {"REDAY_Q", "PEND_STATE", "EXIT_STATE"};
+
+static void event_logger_thread(void)
+{
+	u32_t event_data[4];
+	u16_t event_id;
+	u8_t dropped;
+	u8_t event_data_size = (u8_t)ARRAY_SIZE(event_data);
+	int ret;
+
+	for (;;) {
+
+		ret = sys_k_event_logger_get_wait(&event_id,
+						  &dropped,
+						  event_data,
+						  &event_data_size);
+		if (ret < 0) {
+			continue;
+		}
+
+		timestamp = event_data[0];
+
+		switch (event_id) {
+		case KERNEL_EVENT_LOGGER_CONTEXT_SWITCH_EVENT_ID:
+			printk("tid of context switched thread = %x at time = %d\n"
+				, event_data[1], timestamp);
+			break;
+		case KERNEL_EVENT_LOGGER_INTERRUPT_EVENT_ID:
+			break;
+		case KERNEL_EVENT_LOGGER_SLEEP_EVENT_ID:
+			break;
+		case KERNEL_EVENT_LOGGER_THREAD_EVENT_ID:
+			printk("thread = %x, is moved to = %s ,
```

Comment:
reformat this message, this should fit in 2 lines and would be more readable.

Reply:
ok.